neuralpp/symbolic/basic_expression.py

```diff
@@ -55,7 +55,7 @@ class BasicAtomicExpression(BasicExpression, AtomicExpression, ABC):
     def __init__(self, atom: Any, expression_type: Optional[ExpressionType] = None):
         if expression_type is None and not isinstance(atom, ExpressionType):
             # try to infer type for atom
-            if isinstance(atom, Callable):
+            if callable(atom):
```

Comment:
`isinstance(..., Callable)` and `isinstance(...,type(Callable[...]))` are different (see later comments). This is an equivalent expression of the former, I made this change to avoid confusion with the latter.